src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.js

```diff
@@ -1,23 +1,56 @@
-import React from "react";
+import React, { useEffect, useRef, useState } from "react";
 import PropTypes from "prop-types";
 import Chip from "../../Chip";
 
 import "./filter-panel-section.scss";
 
-const FilterPanelSection = ({ data }) => (
-  <div className="filter-panel-section">
-    {data?.heading && (
-      <h3 className="filter-panel-section__heading">{data.heading}</h3>
-    )}
-    {data?.chips?.map((chip) => (
-      <Chip
-        lead={chip.lead}
-        value={chip.value}
-        key={`${chip.lead}+${chip.value}`}
-      />
-    ))}
-  </div>
-);
+const FilterPanelSection = ({ data }) => {
+  const { chips, heading } = data;
+  const [overflowCounter, setOverflowCounter] = useState(0);
+  const chipWrapper = useRef(null);
+  const viewportWidth = window.innerWidth;
+
+  // if the offsetTop is more than double height of a single chip, consider it
+  // oveflowing
+  const updateFlowCount = function () {
+    const chips = chipWrapper.current.querySelectorAll(".p-chip");
+    let overflowChips = 0;
+    chips.forEach((chip) => {
+      if (chip.offsetTop > chip.offsetHeight * 2) overflowChips++;
+    });
+    setOverflowCounter(overflowChips);
+  };
+
+  useEffect(() => {
+    updateFlowCount();
+    window.addEventListener("resize", updateFlowCount);
+    return () => {
+      window.removeEventListener("resize", updateFlowCount);
+    };
+  }, [chipWrapper, viewportWidth]);
```

Comment:
You're updating this on viewport width but it's possible that the width of this could change on its parent container resizing not just the viewport...

Reply:
Now using resizeObserver to watch size changes in the chips wrapper 👌🏻